### PR TITLE
nodejs 15.6.0

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by minrk** [comment]
Got past the ninja conflict with a patch, but now [failing](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=279791&view=logs&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a&t=f7bcb349-6dca-53ba-d88d-1d9963e119ba&l=1612) with:

<details>

```
135/2302] LIBTOOL-STATIC obj.host/tools/v8_gypfiles/libtorque_base.a, POSTBUILDS
[136/2302] STAMP obj.host/tools/v8_gypfiles/torque.actions_depends.stamp
[137/2302] CXX obj.host/deps/v8/src/builtins/bytecode_builtins_list_generator.generate-bytecodes-builtins-list.o
[138/2302] CXX obj.host/deps/v8/src/interpreter/bytecode_builtins_list_generator.bytecodes.o
[139/2302] CXX obj.host/deps/v8/src/torque/torque.torque.o
[140/2302] LINK torque, POSTBUILDS
FAILED: torque 
$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,$PREFIX/lib -L$PREFIX/lib -Wl,-search_paths_first -mmacosx-version-min=10.13 -L./ -stdlib=libc++ -o torque obj.host/deps/v8/src/torque/torque.torque.o obj.host/tools/v8_gypfiles/libtorque_base.a obj.host/tools/v8_gypfiles/libv8_libbase.a  -L$PREFIX/lib -lz -luv -lcrypto -lssl -licui18n -licuuc -licudata
ld: warning: ignoring file $PREFIX/lib/libz.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libuv.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libcrypto.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libssl.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libicui18n.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libicuuc.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libc++.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file $PREFIX/lib/libicudata.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
```

</details>

The issue being that torque *apparently* needs to be built for both the host and target, which complicates things as dependencies must be linked from both the host and build prefixes.

**Comment by minrk** [comment]
btw, I think it's okay to merge these to clear the backlog of releases and finish fixing arm in a later PR. Is that okay?

**Comment by isuruf** [comment]
You might need to patch in https://github.com/nodejs/node/pull/37444/files

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,18 +28,23 @@ requirements:
...
     - git  # [linux]
     - ninja  # [not win]
     - pkg-config  # [not win]
+
+    # need host builds of several lib dependencies
+    # when cross-compiling
+    - icu  # [osx and arm64]
+    - libuv  # [osx and arm64]
+    - openssl  # [osx and arm64]
+    - zlib  # [osx and arm64]
```
Comment (xhochy) [inline recipe/meta.yaml:41]:
```suggestion
    - icu  # [build_platform != target_platform]
    - libuv  # [build_platform != target_platform]
    - openssl  # [build_platform != target_platform]
    - zlib  # [build_platform != target_platform]
```

**Comment by xhochy** [comment]
@minrk @isuruf Thanks! This is passing now :) I did a small style fix regarding cross-compiling but from my side, I think we are ready for a merge. Any objections?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,14 +8,15 @@ package:
...
-  sha256: cadfa384a5f14591b84ce07a1afe529f28deb0d43366fb0ae4e78afba96bfaf2  # [not win]
+  url: https://nodejs.org/dist/v{{ version }}/node-v{{ version }}.tar.gz       # [not win]
+  sha256: e08c8778a6b767cd0d2ab4613946d1616c3d4054dead08b04f705d607c84510f     # [not win]
   url: https://nodejs.org/dist/v{{ version }}/node-v{{ version }}-win-x64.zip  # [win]
-  sha256: 757c3f5617d28f1fc56ccb8dd7008ea865e482ae9baf5e4bf4d003afa8ae9c8b  # [win]
+  sha256: edb16def1fecaecaffcccf9352e9f94a029d47203f26ae44dc1d4aa7eef35940     # [win]
   patches:
     - linux-librt.patch  # [not win]
     - cinttypes.patch  # [linux]
+    - less-shared-intermediate.patch  # [osx and arm64]
```
Comment (isuruf) [inline recipe/meta.yaml:18]: Do the same here?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -8,14 +8,15 @@ package:
...
+    - ldflags_host.patch  # [osx and arm64]
```
Comment (xhochy) [inline recipe/meta.yaml:19]:
```suggestion
    - less-shared-intermediate.patch  # [build_platform != target_platform]
    - ldflags_host.patch  # [build_platform != target_platform]
```

**Review by minrk — APPROVED** [review]
Yay! Team effort. LGTM.

**Pull request merged**